ui/ui_streamlit.py

```diff
@@ -0,0 +1,1096 @@
+                step=st.session_state['secondsInterval'],
+                min_value=0.0,
+                max_value=max_frames / video_fps,
+                key='secondsNumber'
+            )
+
+
+            def next_frame_button_on_click():
+                st.session_state['frameNumber'] += frameInterval
+                st.session_state['secondsNumber'] += secondsInterval
+
+
+            nextFrameButton = st.button(
+                'Next frame',
+                on_click=next_frame_button_on_click
+            )
+
+            secondsOfVideoPlayed = frameNumber / video_fps
+
+            if st.session_state.get('annotationType') == EVENT_ANNOTATION:
+                with uiColumns[1]:
+                    st.video(
+                        os.path.join(matchDirectory, videoFileName),
+                        start_time=int(secondsOfVideoPlayed)
+                    )
+
+        elif videoModeType == 'By annotations':
+            annotationSecond = st.select_slider(
+                'Select second with annotations',
+                st.session_state[st.session_state['annotationType']].keys(),
+                key='annotationSecond'
+            )
+
+
+            def next_annotation_button_on_click():
+                current_annotations = list(st.session_state[annotationType].keys())
+                current_annotation_index = current_annotations.index(annotationSecond)
+                if current_annotation_index < len(current_annotations) - 1:
+                    next_annotation = current_annotations[current_annotation_index + 1]
+                    st.session_state['annotationSecond'] = next_annotation
+
+
+            nextAnnotationButton = st.button(
+                'Next annotation',
+                on_click=next_annotation_button_on_click
+            )
+
+            secondsOfVideoPlayed = float(annotationSecond)
+
+    secondsRoundedStr = str(float(secondsOfVideoPlayed))
+
+    with uiColumns[2]:
+        annotationType = st.radio(
+            'Choose annotation type',
+            [
+                EVENT_ANNOTATION,
+                FIELD_ANNOTATION,
+                LINE_ANNOTATION,
+                PLAYER_ANNOTATION,
+                BALL_ANNOTATION
+            ],
+            index=0,
+            horizontal=True,
+            key='annotationType'
+        )
+        if annotationType == FIELD_ANNOTATION:
+            canvasDrawingMode = 'polygon'
+            annotations = fieldAnnotations
+        elif annotationType == LINE_ANNOTATION:
+            canvasDrawingMode = 'line'
+            annotations = lineAnnotations
+        elif annotationType == PLAYER_ANNOTATION:
+            canvasDrawingMode = 'rect'
+            annotations = playerAnnotations
+        elif annotationType == BALL_ANNOTATION:
+            canvasDrawingMode = 'rect'
+            annotations = ballAnnotations
+        elif annotationType == EVENT_ANNOTATION:
+            annotations = eventAnnotations
```

Comment:
maybe you could use dict with keys [FIELD_ANNOTATION], values: canva and annotations values
Then you could get rid of all elifs